projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -57,15 +57,46 @@
 int32_t i2c_init(struct i2c_desc **desc,
 		 const struct i2c_init_param *param)
 {
-	if (desc) {
-		// Unused variable - fix compiler warning
-	}
+	i2c_desc *dev;
+	int32_t ret;
 
-	if (param->type) {
-		// Unused variable - fix compiler warning
-	}
+	dev = calloc(1, sizeof *dev);
+	if(!dev)
+		return FAILURE;
+
+	dev->type = param->type;
+	dev->id = XPAR_AXI_IIC_MAIN_DEVICE_ID;
+	dev->max_speed_hz = param->max_speed_hz;
+	dev->slave_address = param->slave_address;
+
+	dev->config = XIic_LookupConfig(dev->id);
+	if (dev->config == NULL)
+		goto error;
+
+	ret = XIic_CfgInitialize(&dev->instance, dev->config,
+				 dev->config->BaseAddress);
+	if(ret != XST_SUCCESS)
```

Comment:
For consistency with the other functions, I would use 0 instead of XST_SUCCESS. Otherwise, I would update similarly the other functions. :)